mule-extensions-api/src/main/java/org/mule/runtime/extension/api/component/ComponentParameterization.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.api.component;
+
+import org.mule.runtime.api.meta.model.parameter.ParameterizedModel;
+import org.mule.runtime.extension.internal.component.ComponentParameterizationBuilder;
+
+/**
+ * Represents a specific configuration for a concrete {@link ParameterizedModel}.
+ * <p>
+ * Instances are validated during creation, so no instances must contain invalid parameters according to the {@link #getModel()
+ * model}.
+ *
+ * @param <M> the actual {@link ParameterizedModel} sub-type of the component .
+ * 
+ * @since 1.5
+ */
+public interface ComponentParameterization<M extends ParameterizedModel> {
+
+  /**
+   * @return the actual model of the declared component.
+   */
+  M getModel();
+
+  /**
+   * Obtains the value of a parameter of the declared component.
+   * 
+   * @param paramGroupName the name of the parameter group
+   * @param paramName      the name of the parameter within the group
+   * @return the value of the parameter.
+   */
+  // TODO W-11214395 determine how complex parameters will be represented.
+  Object getParameter(String paramGroupName, String paramName);
+
+  /**
+   * Iterates through the parameters and calls the provided {@code action} on each one.
+   * 
+   * @param action a callback to be called for every parameter.
+   */
+  void forEachParameter(ParameterAction action);
```

Comment:
we should also have some method to get all parameters in map form.

Reply:
added